tests/app/dao/test_uploads_dao.py

```diff
@@ -0,0 +1,168 @@
+from datetime import datetime, timedelta
+
+from app.dao.uploads_dao import dao_get_uploads_by_service_id
+from app.models import LETTER_TYPE, JOB_STATUS_IN_PROGRESS
+from tests.app.db import create_job, create_service, create_template, create_notification
+
+
+def create_uploaded_letter(letter_template, service, status='created', created_at=None):
+    return create_notification(
+        template=letter_template,
+        to_field="file-name",
+        status=status,
+        reference="dvla-reference",
+        client_reference="file-name",
+        one_off=True,
+        created_by_id=service.users[0].id,
+        created_at=created_at
+    )
+
+
+def create_precompiled_template(service):
```

Comment:
Optional: We could make these two helper methods more consistent in their naming e.g.
`create_uploaded_letter` and `create_uploaded_letter_template`
or
`create_precompiled_letter` and `create_precompiled_template`